File: src/main/java/gargoyle/ct/convert/impl/EnumConverter.java
package gargoyle.ct.convert.impl;

import gargoyle.ct.convert.Converter;
import org.jetbrains.annotations.NotNull;

public class EnumConverter<E extends Enum<E>> implements Converter<E> {
    private final Class<E> type;

    public EnumConverter(Class<E> type) {
        this.type = type;
    }

    @Override
    public String format(@NotNull E data) {
        return data.name();
    }

    @NotNull
    @Override
    public E parse(@NotNull String data) {
        return Enum.valueOf(type, data);
    }
}

File: src/main/kotlin/gargoyle/ct/convert/impl/EnumConverter.kt
package gargoyle.ct.convert.impl

import gargoyle.ct.convert.Converter
import kotlin.reflect.KClass

class EnumConverter<E : Enum<E>>(private val type: KClass<E>) : Converter<E> {
    override fun format(data: E): String {
        return data.name
    }

    override fun parse(data: String): E {
        return type.java.enumConstants.first { it.name == data }
    }
}
